frontend/src/features/admin/agencies/EditAgencyPage.tsx

```diff
@@ -0,0 +1,215 @@
+    isDisabled: agency.isDisabled,
+    sendEmail: agency.sendEmail,
+    email: agency.email,
+    rowVersion: agency.rowVersion,
+  };
+
+  const newValues: any = {
+    code: '',
+    name: '',
+    description: '',
+    isDisabled: false,
+    sendEmail: true,
+    email: '',
+    rowVersion: '',
+  };
+
+  return (
+    <div>
+      {showDelete && <DeleteModal {...{ showDelete, setShowDelete, history, dispatch, agency }} />}
+      <Navbar className="navBar" expand="sm" variant="light" bg="light">
+        <Navbar.Brand href="#">
+          {' '}
+          <FaArrowAltCircleLeft onClick={goBack} size={20} /> Agency Information
+        </Navbar.Brand>
+      </Navbar>
+      <Container fluid={true}>
+        <Row className="agency-edit-form-container">
+          <Formik
+            enableReinitialize
+            initialValues={newAgency ? newValues : initialValues}
+            onSubmit={async (values, { setSubmitting }) => {
+              if (!newAgency) {
+                dispatch(
+                  getUpdateAgencyAction(
+                    { id: agencyId },
+                    {
+                      id: agency.id,
+                      name: values.name,
+                      code: values.code,
+                      email: values.email,
+                      isDisabled: values.isDisabled,
+                      sendEmail: values.sendEmail,
+                      parentId: Number(values.parentId),
+                      description: values.description,
+                      rowVersion: values.rowVersion,
+                    },
+                  ),
+                );
+              } else {
+                const data = await createAgency({
+                  name: values.name,
+                  code: values.code,
+                  email: values.email,
+                  isDisabled: values.isDisabled,
+                  sendEmail: values.sendEmail,
+                  parentId: Number(values.parentId),
+                  description: values.description,
+                })(dispatch);
+                dispatch(clear(actionTypes.ADD_PARCEL));
+                history.replace(`/admin/agency/${data.id}`);
+              }
+              setSubmitting(false);
+            }}
+          >
+            {props => (
+              <Form className="agencyInfo">
+                <Label>Agency</Label>
+                <Input data-testid="agency" field="name" value={props.values.name} type="text" />
+                <Label>Short Name (Code)</Label>
+                <Input data-testid="shortName" field="code" value={props.values.code} type="text" />
+                {checkAgencies}
+                <Label>Description</Label>
+                <Input field="description" type="text" />
+                <Label>Agency e-mail address</Label>
+                <Input data-testid="email" field="email" type="text" />
+
+                <Form.Group className={'is-disabled'}>
+                  <Form.Label>
+                    Is Disabled?{' '}
```

Comment:
Slight preference for getting rid of these spaces after the label text in favor of css for cross browser consistency.

Reply:
Will do